crates/symbolicator-service/src/services/sourcemap_lookup.rs

```diff
@@ -276,19 +272,28 @@ impl SourceMapLookup {
 
 /// Joins a `url` to the `base` [`Url`], taking care of our special `~/` prefix that is treated just
 /// like an absolute url.
-fn join_url(base: &Url, url: &str) -> Option<Url> {
+pub fn join_paths(base: &str, url: &str) -> Option<String> {
+    if SCHEMA_RE.is_match(url) {
+        return Some(url.to_string());
+    }
+
     let url = url.strip_prefix('~').unwrap_or(url);
-    base.join(url).ok()
+    let mut result: PathBuf = base.into();
+    result.pop();
+    result.push(url);
+    let result = result.display().to_string();
+    Some(clean_path(&result).to_string())
 }
 
+// TODO: Rename this
```

Comment:
why exactly? because both are just `&str` and you can always just use `starts_with` instead of a `match`?